core/src/main/java/org/jdbi/v3/ConcreteStatementContext.java

```diff
@@ -89,11 +98,16 @@ public Object getAttribute(String key)
     }
 
     @Override
-    public <T> ResultColumnMapper<T> columnMapperFor(Class<T> type)
+    public ResultColumnMapper<?> columnMapperFor(Type type)
```

Comment:
I'm ok with making type safety promises as long as there's a parameter to justify it:

    default <T> ResultColumnMapper<T> columnMapperFor(GenericType<T> genericType) {
      return (ResultColumnMapper<T>) columnMapperFor(genericType.getType());
    }
    default <T> ResultColumnMapper<T> columnMapperFor(Class<T> clazz) {
      return (ResultColumnMapper<T>) columnMapperFor((Type) clazz);
    }
    ResultColumnMapper<?> columnMapperFor(Type)

This means that factories are on the hook for type correctness--which is what we expect from them anyway.